tests/scripts/thread-cert/Cert_5_3_06_RouterIdMask.py

```diff
@@ -89,64 +113,97 @@ def test(self):
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
 
         # 6
         self.nodes[ROUTER1].reset()
         self.nodes[ROUTER2].reset()
 
-        router1_id = self.nodes[ROUTER1].get_router_id()
-        router2_id = self.nodes[ROUTER2].get_router_id()
-
         self.simulator.go(config.MAX_NEIGHBOR_AGE + config.MAX_ADVERTISEMENT_INTERVAL)
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
-        self.assertEqual(command.get_routing_cost(msg, router1_id), 0)
 
-        self.simulator.go(config.INFINITE_COST_TIMEOUT + config.MAX_ADVERTISEMENT_INTERVAL)
-        leader_messages = self.simulator.get_messages_sent_by(DUT_LEADER)
-        msg = leader_messages.last_mle_message(mle.CommandType.ADVERTISEMENT)
-        self.assertFalse(command.check_id_set(msg, router1_id))
-        self.assertFalse(command.check_id_set(msg, router2_id))
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+
+        # Step 1: Ensure topology is formed correctly
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(LEADER).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_wpan_dst64(ROUTER_1).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
+        pkts.filter_wpan_src64(ROUTER_2).\
+            filter_wpan_dst64(ROUTER_1).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            must_next()
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_wpan_dst64(ROUTER_2).\
+            filter_mle_cmd(MLE_CHILD_ID_RESPONSE).\
+            must_next()
+
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,2,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+
+        # Step 4: The DUT’s routing cost to Router_2 MUST count to infinity
+        #         The DUT MUST remove Router_2 ID from its ID set
+        #         Verify route data has settled
+        _pkt = pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,0,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+        pkts.filter_wpan_src64(LEADER).\
+            filter_LLANMA().\
+            filter_mle_cmd(MLE_ADVERTISEMENT).\
+            filter(lambda p: {1,1} == set(p.mle.tlv.route64.cost)).\
+            must_next()
+
+        # Step 5: Re-attach Router_2 to Router_1.
+        #         The DUT MUST reset the MLE Advertisement trickle timer and
```

Comment:
Check the trickle timer is reset?
Maybe by inspecting two continuous MLE advertisement messages from DUT in short time (like no more than 3s)?

Reply:
added the check